web-common/src/features/dashboards/pivot/PivotExpandableRow.svelte

```diff
@@ -0,0 +1,113 @@
+<script lang="ts">
+  import type { Row, Cell } from "@tanstack/svelte-table";
+  import type { PivotDataRow } from "./types";
+  import ChevronRight from "@rilldata/web-common/components/icons/ChevronRight.svelte";
+  import { pivotFormatter } from "@rilldata/web-common/lib/number-formatting/format-measure-value";
+
+  export let row: Row<PivotDataRow>;
+  export let measureCount: number;
+  export let hasDimension: boolean;
+  export let totals = false;
+
+  $: cells = row.getVisibleCells() as Cell<
+    PivotDataRow,
+    string | number | undefined
+  >[];
+  $: rowHeader = cells[0].getValue() as string | null;
+  $: canExpand = row.getCanExpand();
+  $: expanded = row.getIsExpanded();
+</script>
+
+<tr class:with-row-dimension={hasDimension} class:totals>
+  {#if hasDimension}
+    <td class="row-header">
+      <svelte:element
+        this={canExpand ? "button" : "div"}
+        role={canExpand ? "button" : undefined}
+        class="cell gap-x-1 flex items-center w-full"
+        class:font-normal={row.depth >= 1}
+        on:click={canExpand ? () => row.toggleExpanded() : undefined}
+      >
+        {#if rowHeader === "LOADING_CELL"}
+          <span class="loading-cell" />
+        {:else}
+          <div
+            class="gap-x-1 flex items-center w-full"
+            style:padding-left="{row.depth * 1.5}rem"
+          >
+            {#if canExpand}
+              <span class="transition-transform" class:rotate-90={expanded}>
+                <ChevronRight />
+              </span>
+            {/if}
+            {rowHeader}
+          </div>
+        {/if}
+      </svelte:element>
+    </td>
+  {/if}
+
+  {#each hasDimension ? cells.slice(1) : cells as cell, i (cell.id)}
+    {@const value = cell.getValue()}
+    {@const format = String(cell.column.columnDef.cell)}
+    <td class:border-r={i && (i + 1) % measureCount === 0}>
```

Comment:
There's a line missing on the right of row totals

![image](https://github.com/rilldata/rill/assets/4402679/d7dddd12-a19f-4f89-9c8e-4efd74fb987f)